File: 1 Array & Hashing/3 Two Sum/Solution.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    for (int first = 0 ; first < nums.size() - 1; first++) {
        for (int second = first + 1 ; second < nums.size() ; second++) {
            if (nums[first] + nums[second] == target) {
                vector<int> result = {first, second};
                return result;
            }
        }
    }
    vector<int> result = {-1, -1};
    return result;
}

int main() {
    vector<int> vec{2, 3, 4};
    vector<int> result = twoSum(vec, 9);

    for(int item: result) {
        cout << item << " ";
    }
    return 0;
}

File: 1 Array & Hashing/3 Two Sum/Solution.py
def twoSum(nums: list[int], target: int) -> list[int]:
        for i in range(len(nums)) :
            if (nums[i] <= target) :
                for j in range(i + 1, len(nums)) :
                    print(nums[i], nums[j])
                    if (nums[i] + nums[j] == target) :
                        return [i, j]
                    
if __name__ == "__main__" :
    print(twoSum([1, 2, 3, 4, 5, 6], 6))